### ISSUE TITLE
[BUG] Object has no attribute 'group'

### ISSUE THREAD

**Issue by LucasRoot** [nishad/udemy-dl-windows#95]
Version : **2.1.1**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/nishad/udemy-dl-windows/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed tab](https://github.com/nishad/udemy-dl-windows/issues?q=is%3Aissue+is%3Aclosed)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [X] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

**If the project maintainer has any additional requirements, they will be listed here.**

- No additional requirements.

---

Hello, whenever I try to download a course, this happens: "'NoneType' object has no a
ttribute 'group'"
Here is the debug of the script udemy-dl.exe
```
`[INFO-881] Downloading to: C:\Users\Windows\Downloads\udemy-dl-win-2.1.1\udemy-d
l-win\erfolgreich-im-job-beste-excel-ninja-tricks

[INFO-127] Trying to log in ...
[INFO-157] Login success.
[INFO-189] Found course id: 592162
Getting course data: 12.00% [==------------------] [CRITICAL-258] Skipped. Expec
ted quality not found!
Getting course data: 16.00% [===-----------------] [CRITICAL-425] Cannot downloa
d lecture "Überflüssige Informationen fliegen raus": "'NoneType' object has no a
ttribute 'group'"
Getting course data: 20.00% [====----------------] [CRITICAL-425] Cannot downloa
d lecture "Zuerst einmal ausloten wo wir sind.": "'NoneType' object has no attri
bute 'group'"
Getting course data: 24.00% [=====---------------] [CRITICAL-425] Cannot downloa
d lecture "Total unterbewertet im Berufsalltag Lücken auffüllen": "'NoneType' ob
ject has no attribute 'group'"
Getting course data: 32.00% [======--------------] [CRITICAL-425] Cannot downloa
d lecture "Überraschend oft müssen wir Dinge vergleichen": "'NoneType' object ha
s no attribute 'group'"
Getting course data: 36.00% [=======-------------] [CRITICAL-425] Cannot downloa
d lecture "Wie man Unterschiede innerhalb von Sekunden findet": "'NoneType' obje
ct has no attribute 'group'"
Getting course data: 40.00% [========------------] [CRITICAL-425] Cannot downloa
d lecture "Delta-Analysen bitte immer in beide Richtungen machen": "'NoneType' o
bject has no attribute 'group'"
Getting course data: 44.00% [=========-----------] [CRITICAL-425] Cannot downloa
d lecture "Ganz wichtig Wie Du die Analyse-Email richtig aufsetzt und formuliers
t": "'NoneType' object has no attribute 'group'"
Getting course data: 52.00% [==========----------] [CRITICAL-425] Cannot downloa
d lecture "Philosophischer Auftakt zu SVERWEIS": "'NoneType' object has no attri
bute 'group'"
Getting course data: 56.00% [===========---------] [CRITICAL-425] Cannot downloa
d lecture "Wie man SVERWEIS wirklich richtig verwendet": "('Connection aborted.'
, error(10054, 'Foi for\xe7ado o cancelamento de uma conex\xe3o existente pelo h
ost remoto'))"
Getting course data: 60.00% [============--------] [CRITICAL-425] Cannot downloa
d lecture "Wie man Zahlen innerhalb einer Spalte automatisch aktualisiert": "'No
neType' object has no attribute 'group'"`
```
Thanks a lot!

**Comment by LucasRoot** [comment]
Here is the image for easily understand: http://i.imgur.com/Yswux0c.png

**Comment by LucasRoot** [comment]
Sorry, forgot to check the version, the version is v2.1.1

**Comment by kerb9729** [comment]
I am experiencing the same issue.  Some videos in my course (unitycourse) do download.
The ones which do download seem to be newer videos.  The instructor updates course material fairly frequently.

My debug output looks similar to that which is already attached here.  What can I do to help troubleshoot the issue?

**Comment by luizconrado** [comment]
I am also having the same exact problem. I would be great if you could fix it. My course is:

**Comment by Sieabah** [comment]
I submitted a solution https://github.com/nishad/udemy-dl-windows/pull/98

**Comment by theglowingnation** [comment]
Is there any final solution for this problem ? Please help

**Comment by glasgreat** [comment]
@theglowingnation 

I found an alternative solution, and that is using YouTube-dl it also works for Udemy downloads.

https://github.com/rg3/youtube-dl/blob/master/README.md#readme

**Comment by loisishere** [comment]
I get the same errors when downloading videos. I did the debug for udemy-dl and youtube-dl
![image](https://cloud.githubusercontent.com/assets/19920534/18220716/d9bd388a-7142-11e6-8f4e-35d68b0c795c.png)

How do I fix this? I have v2.2.1

Thanks!

**Comment by rabih96** [comment]
@loisishere Download youtube-dl.exe from here https://github.com/rg3/youtube-dl/blob/master/README.md#readme
Then use youtube-dl.exe -u [USERNAME/EMAIL] -p [PASSWORD] [COURSE URL]
worked for me.

**Comment by KingKongBanana** [comment]
the youtube thing works 
![udemy downloads](https://cloud.githubusercontent.com/assets/15786524/18891662/8a948696-8506-11e6-838b-02b97f318923.JPG)

**Comment by buoing747** [comment]
I have the same issue recently. In addition, as I was downloading the list of the course videos by internet download manager; after finishing about half of the videos, it stopped and it's asking for user & password which keeps asking the same after entering my credentials.

Please help me with that.

**Comment by nishad** [comment]
This is already addressed and fixed with v2.1.2. Try with the latest release https://github.com/nishad/udemy-dl-windows/releases/tag/v2.1.2
